lib/addons/jupyterhub/index.ts

```diff
@@ -0,0 +1,52 @@
+import { Construct } from "constructs";
+import { ClusterInfo } from '../../spi';
+import { createNamespace } from '../../utils';
+import { HelmAddOn, HelmAddOnProps, HelmAddOnUserProps } from '../helm-addon';
+
+/**
+ * Configuration options for the add-on.
+ */
+ type JupyterHubAddOnProps = HelmAddOnUserProps;
+
+const JUPYTERHUB = 'jupyterhub';
+const RELEASE = 'blueprints-addon-jupyterhub';
+
+/**
+ * Defaults options for the add-on
+ */
+const defaultProps: HelmAddOnProps = {
+    name: JUPYTERHUB,
+    namespace: JUPYTERHUB,
+    version: '1.2.0',
+    chart: JUPYTERHUB,
+    release: RELEASE,
+    repository: 'https://jupyterhub.github.io/helm-chart/',
+};
+
+/**
+ * Implementation of the JupyterHub add-on
+ */
+export class JupyterHubAddOn extends HelmAddOn {
```

Comment:
Based on my understanding how the notebooks work it is seems to be a stateful app that relies on storage to store notebooks. Don't we need a dependency on the csi storage driver e.g. for EBS?

Reply:
The storage can be either EBS or EFS, which makes this dependency a bit more difficult to enforce.

Reply:
Once I can add attributes for EFS as data store, I will implement the EBS or EFS CSI addon as a dependency for jupyterhub

Reply:
Resolved as EBS added - EFS will be added as a followup in a future PR.